src/pal/inc/pal.h

```diff
@@ -6414,31 +6418,111 @@ PALAPI
 PAL_SetHardwareExceptionHandler(
     IN PHARDWARE_EXCEPTION_HANDLER exceptionHandler);
 
-class PAL_CatchHardwareExceptionHolder
+//
+// This holder is used to indicate that a hardware
+// exception should be raised as a C++ exception
+// to better emulate SEH on the xplat platforms.
+//
+class CatchHardwareExceptionHolder
 {
 public:
-    PAL_CatchHardwareExceptionHolder();
+    CatchHardwareExceptionHolder();
 
-    ~PAL_CatchHardwareExceptionHolder();
+    ~CatchHardwareExceptionHolder();
 
     static bool IsEnabled();
 };
 
 #ifdef FEATURE_ENABLE_HARDWARE_EXCEPTIONS
-#define HardwareExceptionHolder PAL_CatchHardwareExceptionHolder __catchHardwareException;
+#define HardwareExceptionHolder CatchHardwareExceptionHolder __catchHardwareException;
 #else
 #define HardwareExceptionHolder
 #endif // FEATURE_ENABLE_HARDWARE_EXCEPTIONS
 
 #ifdef FEATURE_PAL_SXS
 
+extern "C++" {
+
+//
+// This is the base class of native exception holder used to provide 
+// the filter function to the exception dispatcher. This allows the
+// filter to be called during the first pass to better emulate SEH
+// the xplat platforms that only have C++ exception support.
+//
+class NativeExceptionHolderBase : CatchHardwareExceptionHolder
```

Comment:
@mikem8361 Actually, maybe leave it as it is for now. I have pulled this change to my repo and already progressed a lot on the remaining part of the work, so changes here would cause merging conflicts.
We can always change it later if we agree on that.